src/app/main/app.component.ts

```diff
@@ -76,14 +63,20 @@ export class AppComponent implements OnInit {
         return this._router.isActive(route, true);
     }
 
+    onActivate(component) {
+        if (!Object.getOwnPropertyNames(component).find(name => component[name] instanceof BreadcrumbsService)) {
```

Comment:
add some comment for the purpose of this function

Reply:
added